reference/features/security.md

```diff
@@ -0,0 +1,237 @@
+# Security
+
+MeiliSearch gives you fine-grained control over which users can access which indexes, endpoints, and routes. When protecting your instance with a master key, you can ensure only authorized users can carry out sensitive tasks such as adding documents or altering index settings.
+
+In this guide, you will learn how to protect your MeiliSearch instance with a master key and communicate with it using an API key. You will also see how you can use your instance's master key to create, update, list and delete API keys.
+
+## Protecting a MeiliSearch instance
+
+By default, MeiliSearch is unprotected. This means all routes are publicly accessible.
+
+To protect a MeiliSearch instance from unauthorized use, you must supply an alphanumeric string at launch. This alphanumeric string is your master key.
+
+Setting up a master key can be done with either command-line options or environment variables. You can read more about [master key configuration in our instance configuration guide](/reference/features/configuration.md#master-key).
+
+:::: tabs
+::: tab CLI
+
+```sh
+./meilisearch --master-key="[your_master_key]"
+```
+
+:::
+
+::: tab Environment variable
+
+```sh
+MEILI_MASTER_KEY="[your_master_key]"
+./meilisearch
+```
+
+:::
+::::
+
+::: note
+Launching an instance with a master key is mandatory when `env` is set to `production`.
+:::
+
+Once you launch MeiliSearch with a master key, the instance is protected. From that point on, [communicating with a protected instance](#communicating-with-a-protected-instance) requires adding an `Authorization` header to requests.
+
+## Communicating with a protected instance
+
+After an instance is secured, only the [`GET /health` endpoint](/reference/api/health.md) will be publicly available. To access any other API endpoint, you must add a security key with suitable permissions to your request.
+
+It is particularly important to protect an instance when dealing with sensitive data. You can use API keys to ensure only authorized people can search through an index containing medical records:
+
+<CodeSamples id="security_guide_search_key_1" />
+
+MeiliSearch currently has two types of security keys: one master key and any number of API keys.
```

Comment:
I think it's actually possible to say that, yes, both master and API keys are currently two types of the same thing, but, confusingly, no, they are not subject to the same rules. This particular issue has been one of the major pain points in writing this guide for me: "yes, both things are called keys and their functionality overlaps, but they are not the same thing."

Perhaps clarity here is difficult because the spec itself is muddied in how it wants the master key to be used in one specific way, but also tacitly allows people to go wild and use it for everything?